tools/ui-components/src/form-control/form-control.tsx

```diff
@@ -2,58 +2,49 @@ import React from 'react';
 
 import { FormControlFeedback as Feedback } from './form-control-feedback';
 import { FormControlStatic as Static } from './form-control-static';
-import { FormControlProps, FormControlElement } from './types';
+import { FormControlProps } from './types';
 
 // Uses controlId from <FormGroup> if not explicitly specified.
 // type Only relevant if componentClass is 'input'.
 
-const FormControl = React.forwardRef<FormControlElement, FormControlProps>(
-  (
-    {
-      className,
-      id,
-      testId,
-      onChange,
-      value,
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      componentClass: Component = 'input',
-      placeholder,
-      name,
-      required,
-      type
-    },
-    ref
-  ): JSX.Element => {
-    let defaultClasses =
-      'outline-0 block w-full py-1.5 px-2.5 text-md text-foreground-primary ' +
-      'bg-background-primary bg-none rounded-none border-1 border-solid ' +
-      'border-background-quaternary shadow-none ' +
-      'transition ease-in-out duration-150 focus:border-foreground-tertiary';
+const FormControl = ({
+  className,
+  testId,
+  onChange,
+  value,
+  componentClass,
+  placeholder,
+  name,
+  required,
+  type,
+  ...restProps
+}: FormControlProps): JSX.Element => {
+  const defaultClasses =
+    'outline-0 block w-full py-1.5 px-2.5 text-md text-foreground-primary ' +
+    'bg-background-primary bg-none rounded-none border-1 border-solid ' +
+    'border-background-quaternary shadow-none ' +
+    'transition ease-in-out duration-150 focus:border-foreground-tertiary';
+  const Component = componentClass || 'input';
+  let variantClass;
+  if (Component !== 'textarea') variantClass = 'h-8';
 
-    let variantClass;
-    if (Component === 'textarea') variantClass = 'h-auto';
-    else defaultClasses += 'h-8';
+  //row and componentClass
+  const classes = [defaultClasses, variantClass, className].join(' ');
 
-    //row and componentClass
-    const classes = [defaultClasses, variantClass, className].join(' ');
-
-    return (
-      <Component
-        ref={ref}
-        id={id}
-        data-testid={testId}
-        className={classes}
-        value={value}
-        required={required}
-        onChange={onChange}
-        name={name}
-        placeholder={placeholder}
-        type={type}
-      />
-    );
-  }
-);
+  return (
+    <Component
+      data-testid={testId}
+      className={classes}
+      value={value}
+      required={required}
+      onChange={onChange}
+      name={name}
+      placeholder={placeholder}
+      type={type}
+      {...restProps}
```

Comment:
removed `id` because the component takes HTML attributes from `React.HTMLAttributes<FormControlElement>` and it will conflict with `{...restProps}`.